src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -178,13 +179,77 @@ SettingsLoader::SettingsLoader(const std::string_view& userJSON, const std::stri
 // (meaning profiles specified by the application rather by the user).
 void SettingsLoader::GenerateProfiles()
 {
-    _executeGenerator(PowershellCoreProfileGenerator{});
-    _executeGenerator(WslDistroGenerator{});
-    _executeGenerator(AzureCloudShellGenerator{});
-    _executeGenerator(VisualStudioGenerator{});
+    PowershellCoreProfileGenerator powerShellGenerator{};
+    _executeGenerator(powerShellGenerator);
+
+    WslDistroGenerator wslGenerator{};
+    _executeGenerator(wslGenerator);
+
+    AzureCloudShellGenerator acsGenerator{};
+    _executeGenerator(acsGenerator);
+
+    VisualStudioGenerator vsGenerator{};
+    _executeGenerator(vsGenerator);
+
 #if TIL_FEATURE_DYNAMICSSHPROFILES_ENABLED
-    _executeGenerator(SshHostGenerator{});
+    SshHostGenerator sshGenerator{};
+    _executeGenerator(sshGenerator);
 #endif
+
+    PowershellInstallationProfileGenerator pwshInstallationGenerator{};
+    _executeGenerator(pwshInstallationGenerator);
+
+    _cleanupPowerShellInstaller(!powerShellGenerator.GetPowerShellInstances().empty());
+}
+
+// Retrieve the "Install Latest PowerShell" profile and...
+// - add a comment to the JSON to indicate it's conditionally applied
+// - (if PowerShell is installed) mark it for deletion
+void SettingsLoader::_cleanupPowerShellInstaller(bool isPowerShellInstalled)
+{
+    const hstring pwshInstallerNamespace{ PowershellInstallationProfileGenerator::Namespace };
+    if (extensionPackageMap.contains(pwshInstallerNamespace))
+    {
+        if (const auto& fragExtList = extensionPackageMap[pwshInstallerNamespace]->Fragments(); fragExtList.Size() > 0)
+        {
+            Json::StreamWriterBuilder styledWriter;
+            styledWriter["indentation"] = "    ";
+            styledWriter["commentStyle"] = "All";
+
+            auto fragExt = get_self<FragmentSettings>(fragExtList.GetAt(0));
+
+            // We want the comment to be the first thing in the object,
+            // "closeOnExit" is the first property, so target that.
+            auto fragExtJson = _parseJSON(til::u16u8(fragExt->Json()));

```

Comment:
To be honest, I kind of forgot how the extension code works, but this still seems kind of hacky to me.

The fragment in this case is based on an extension, right? Don't extensions generate profiles only? I'm missing the big picture how we go from profiles to JSON here, hence my confusion. Basically, since extensions generate profiles (including `PowershellInstallationProfileGenerator`), my assumption would be that this function doesn't need to deal with parsing JSON at all.

Reply:
Oh this section is SUPER hacky! The sole purpose of this is to add the JSON comment so it appears like this:
![image](https://github.com/user-attachments/assets/bea6f0fe-9412-4e49-980c-53b217553778)
![image](https://github.com/user-attachments/assets/7dc78f16-deeb-464f-995d-9be1137cb0bc)

So yeah, we have to parse the JSON, add the comment, then store the new JSON with the comment back in.
We have to do it twice:
1. the settings.json blob representing the entire fragment extension
2. the generated profile itself

Reply:
Figured it'd be good to try and tell the user that this profile only appears if PowerShell is not installed. It's a weird little behavior that I figured at least a JSON comment would help point out, even if not many people read the JSON comment.